memcached.c

```diff
@@ -2914,6 +2914,33 @@ typedef struct token_s {
 
 #define MAX_TOKENS 24
 
+#define WANT_TOKENS(ntokens, min, max) \
+    do { \
+        if (min != -1 && ntokens < min) { \
+            out_string(c, "CLIENT_ERROR wrong number of command tokens"); \
+            return; \
+        } else if (max != -1 && ntokens > max) { \
+            out_string(c, "CLIENT_ERROR wrong number of command tokens"); \
+            return; \
+        } \
+    } while (0)
+
+#define WANT_TOKENS_OR(ntokens, a, b) \
```

Comment:
we need this macro now?
Anything implemented with this macro can be replaced with WANT_TOKENS.
For example, WANT_TOKENS_OR(ntokens, 6, 7) == WANT_TOKENS(ntokens, 6, 7)

Reply:
those're all new. in my first pass I had everything with just WANT_TOKENS(), but most of the paths were originally just two ||'s or a single >= so I added the extra two macros to not add significant slowdown.